### PR TITLE
feat: add goldenfile sanitizers

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds sanitization functionality which allows modification of goldenfile data. A few common sanitizers are included in the package for helping us scrub values that change between test runs.

### CODE REVIEW

File: infra/blueprint-test/pkg/golden/golden.go [hunk 1]
```diff
@@ -37,9 +38,26 @@ const (
...
+
+// StringSanitizer replaces all occurrences of old string with new string
+func StringSanitizer(old, new string) Sanitizer {
+	return func(s string) string {
+		old := strings.ReplaceAll(s, old, new)
+		return old
+	}
+}
+
+// ProjectIDSanitizer replaces all occurrences of current gcloud project ID with PROJECT_ID string
```
Comment (morgante) [inline infra/blueprint-test/pkg/golden/golden.go:57]: Should we include this by default?

File: infra/blueprint-test/pkg/golden/golden.go [hunk 2]
```diff
@@ -91,6 +118,21 @@ func (g *GoldenFile) GetName() string {
...
+// ApplySanitizers returns sanitized string
+func (g *GoldenFile) ApplySanitizers(s string) string {
+	for _, sanitizer := range g.sanitizers {
+		s = sanitizer(s)
+	}
+	return s
+}
+
+// ApplySanitizersJSON returns sanitizes and returns JSON result
+func (g *GoldenFile) ApplySanitizersJSON(s gjson.Result) gjson.Result {
```
Comment (morgante) [inline infra/blueprint-test/pkg/golden/golden.go:130]:
```suggestion
func (g *GoldenFile) GetSanitizedJSON (s gjson.Result) gjson.Result {
```

**Review by morgante — CHANGES_REQUESTED** [review]

File: infra/blueprint-test/pkg/golden/golden.go (hunk 2 above)
Comment (bharathkkb) [reply infra/blueprint-test/pkg/golden/golden.go]: Done

File: infra/blueprint-test/pkg/golden/golden.go (hunk 1 above)
Comment (bharathkkb) [reply infra/blueprint-test/pkg/golden/golden.go:57]: I am leaning towards no as we don't set gcloud config in container today. We can add this as a default later on if it becomes a common pattern.

Comment (morgante) [reply infra/blueprint-test/pkg/golden/golden.go:57]: What's the behavior if it's not set? (Can we add a test for that?)

Assuming it doesn't cause any errors if unset, I suggest we use it be default.

Comment (bharathkkb) [reply infra/blueprint-test/pkg/golden/golden.go:56]: @morgante added as default and added a test. If its not set exit code is still 0 with an empty result. I added some additional logging for this case, so users know its skipping.

**Pull request merged**